product-scenarios/2-Collaborative-API-Development/2.1-Manage-publisher-portal-operations-for-users/2.1.1-Manage-API-publishing-according-to-the-publisher-roles-and-permission/src/test/java/org/wso2/am/scenario/tests/rest/api/publisherRoleAndPermission/PublishAPIByValidRoleCategoryTestCase.java

```diff
@@ -0,0 +1,229 @@
+        assertTrue(publishAPI.getData().contains("PUBLISHED"),
+                "API has not been created in publisher");
+    }
+
+    @Test(description = "2.1.1.2", dataProvider = "PermissionValidationDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testPublishAPIByValidPermissionUser(String[] permissionList) throws Exception {
+
+        apiName = "API_" + count;
+        apiContext = "/verify_" + count;
+        creatorUser = "User_" + count;
+        userRole = "role" + count;
+        testUser = "User" + count;
+        password = "password123$";
+        count++;
+
+        createUserWithCreatorRole(creatorUser, password, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(creatorUser, password);
+
+        apiRequest = new APIRequest(apiName, apiContext, apiVisibility, apiVersion, apiResource, tierCollection,
+                new URL(backendEndPoint));
+        createAPI(apiRequest);
+        getAPI(apiName, creatorUser, apiVersion);
+        apiNames.put(apiName, creatorUser);
+
+        createRole(ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD, userRole, permissionList);
+        roleList.add(userRole);
+
+        createUser(testUser, password, new String[]{userRole}, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        userList.add(testUser);
+        apiPublisher.logout();
+
+        apiPublisher.login(testUser, password);
+        HttpResponse publishAPI = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.PUBLISHED);
+        verifyResponse(publishAPI);
+        assertTrue(publishAPI.getData().contains("PUBLISHED"),
+                "API has not been created in publisher");
+    }
+
+    @Test(description = "2.1.1.3", dataProvider = "RoleUpdatingDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testPublishAPIByUpdatingRoleInUser(String role) throws Exception {
+
+        apiName = "API" + count;
+        apiContext = "/verify" + count;
+        creatorUser = "User_" + count;
+        testUser = "User" + count;
+        password = "password123$";
+        count++;
+
+        createUserWithCreatorRole(creatorUser, password, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(creatorUser, password);
+        apiRequest = new APIRequest(apiName, apiContext, apiVisibility, apiVersion, apiResource, tierCollection,
+                new URL(backendEndPoint));
+        createAPI(apiRequest);
+        getAPI(apiName, creatorUser, apiVersion);
+        apiNames.put(apiName, creatorUser);
+        userList.add(testUser);
+        apiPublisher.logout();
+
+        createUser(testUser, password, new String[]{role}, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(testUser, password);
+        HttpResponse publishAPI = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.PUBLISHED);
+        assertFalse(publishAPI.getData().contains("PUBLISHED"), "API has been created in publisher");
```

Comment:
shouldn't the message be "API has been published using a CREATOR role" something?